src/test/scala/com/cognite/sdk/scala/v1/FilesTest.scala

```diff
@@ -132,4 +141,22 @@ class FilesTest extends SdkTest with ReadBehaviours with WritableBehaviors {
       )
     assert(limitTimeSearchResults.length == 5)
   }
+  it should "support upload" in {
+    val res =
+      client.files.uploadWithName(
+        "./src/test/scala/com/cognite/sdk/scala/v1/",
+        "uploadTest.txt"
+      )
+    res match {
+      case Right(file) => client.files.deleteById(file.id)
+      case Left(_) => ()
+    }
+
+    val name = res match {
+      case Left(_) => ""
+      case Right(file) => file.name
+    }
+
+    assert(name == "uploadTest.txt")
```

Comment:
We should also add support for downloading files, https://doc.cognitedata.com/api/v1/#operation/downloadLinks and use it to verify that the upload was done correctly, but we can do that in a separate PR.